### PR TITLE
[change] Do not send email notifications to unverified email addresses

### PR DESCRIPTION
**Pull request by Dhanus3133**
Closes: #270

### CODE REVIEW

File: openwisp_notifications/handlers.py [hunk 1]
```diff
@@ -182,7 +183,12 @@ def send_email_notification(sender, instance, created, **kwargs):
         # therefore send email anyway.
         email_preference = True
 
-    if not (email_preference and instance.recipient.email):
+    email_address = EmailAddress.objects.filter(
+        user=instance.recipient, email=instance.recipient.email
+    ).first()
+    is_email_verified = email_address.verified if email_address else False
```
Comment (nemesifier) [inline openwisp_notifications/handlers.py:189]: I think this can be shortened to just:

```
email_verified = recipient.emailaddress_set.filter(verified=True, email=recipient.email).exists()
```

**Review by nemesifier — CHANGES_REQUESTED** [review]
Thanks @Dhanus3133, see my comments below.

A test is missing from this patch.

Please add a test that sends a notification to an user having an unverified email address and expects zero email sent.
The test should fail on the current master but pass on this branch.

File: openwisp_notifications/handlers.py (hunk 1 above)
Comment (Dhanus3133) [reply openwisp_notifications/handlers.py]: Updated with the changes.

**Review by nemesifier — APPROVED** [review]
